src/state/slices/opportunitiesSlice/opportunitiesApiSlice.ts

```diff
@@ -83,70 +82,73 @@ export const opportunitiesApi = createApi({
         }
       },
     }),
-    getOpportunityMetadata: build.query<GetOpportunityMetadataOutput, GetOpportunityMetadataInput>({
-      queryFn: async ({ opportunityId, defiType, defiProvider }, { dispatch, getState }) => {
-        try {
-          const resolver = getMetadataResolversByDefiProviderAndDefiType(defiProvider, defiType)
-
-          if (!resolver) {
-            throw new Error(`resolver for ${defiProvider}::${defiType} not implemented`)
-          }
-
-          const resolved = await resolver({
-            opportunityId,
-            defiType,
-            reduxApi: { dispatch, getState },
-          })
-
-          dispatch(opportunities.actions.upsertOpportunitiesMetadata(resolved.data))
+    getOpportunityMetadata: build.query<null, GetOpportunityMetadataInput[]>({
+      queryFn: async (queries, { dispatch, getState }) => {
+        await Promise.allSettled(
+          queries.map(async ({ opportunityId, defiType, defiProvider }) => {
+            try {
+              const resolver = getMetadataResolversByDefiProviderAndDefiType(defiProvider, defiType)
+
+              if (!resolver) {
+                console.warn(`resolver for ${defiProvider}::${defiType} not implemented`)
+                return
+              }
+
+              const resolved = await resolver({
+                opportunityId,
+                defiType,
+                reduxApi: { dispatch, getState },
+              })
+
+              // TODO: collect and dispatch once to improve perf locally
+              dispatch(opportunities.actions.upsertOpportunitiesMetadata(resolved.data))
+            } catch (e) {
+              const message = e instanceof Error ? e.message : 'Error getting opportunity metadata'
+              console.error(message)
+            }
+          }),
+        )
 
-          return { data: resolved.data }
-        } catch (e) {
-          const message = e instanceof Error ? e.message : 'Error getting opportunity metadata'
-          return {
-            error: {
-              error: message,
-              status: 'CUSTOM_ERROR',
-            },
-          }
-        }
+        return { data: null }
       },
     }),
     getOpportunitiesMetadata: build.query<
-      GetOpportunityMetadataOutput,
-      Omit<GetOpportunityMetadataInput, 'opportunityId'>
+      null,
```

Comment:
This with the TODO below is fine at the moment since we don't consume the thunk resolved data, but we absolutely want to bring back the response type here, as RTK thunks are conventionally expected to resolve with some data we can access by dispatch awaiting them 👀

Reply:
sounds good i'll resolve in follow-up

Reply:
will also tackle log spew at same time